### PR TITLE
added: prop-types and main components typisation

### PR DESCRIPTION
**Pull request by andriiwp3**

### CODE REVIEW

File: src/components/cabinet/profile-item.component.js [hunk 1]
```diff
@@ -143,4 +145,15 @@ const ProfileItem = ({ profile }) => {
...
 };
 
+ProfileItem.propTypes = {
+  profile: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    mainPhoto: fileType.isRequired,
+    firstName: PropTypes.string.isRequired,
+    lastName: PropTypes.string.isRequired,
+    surName: PropTypes.string,
+    description: PropTypes.string.isRequired,
```
Comment (AndriyTykhan) [inline src/components/cabinet/profile-item.component.js:155]: There are more properties in the profile item, could you please add all of them

File: src/components/error/error.component.js [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
     </div>
-  )
-}
+  );
+};
+
+Error.propTypes = {
+  errorHeader: PropTypes.string.isRequired,
+  errorSubText: PropTypes.string.isRequired,
+  errorBtnText: PropTypes.string.isRequired,
```
Comment (AndriyTykhan) [inline src/components/error/error.component.js:27]: Perhaps let's add default prop types here? 
Just something like: Opps, something goes wrong

File: src/types/profileData.types.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+export const fileType = shape({
+  fileName: string,
+  id: string,
+  key: string,
+  mimeType: string,
+  _id: string,
+});
+
+export const profileDataTypes = shape({
```
Comment (AndriyTykhan) [inline src/types/profileData.types.js:11]: Let's rename it to just `profile`

File: src/components/cabinet/profile-item.component.js (hunk 1 above)
Comment (AndriyTykhan) [reply src/components/cabinet/profile-item.component.js:155]: Why don't use `profileDataTypes` here?

File: src/components/profiles/profiles-card/profiles-card.component.js [hunk 4]
```diff
@@ -25,4 +27,13 @@ const ProfilesCard = ({ profile }) => {
   );
 };
 
+ProfilesCard.propTypes = {
```
Comment (AndriyTykhan) [inline src/components/profiles/profiles-card/profiles-card.component.js:30]: Why don't use `profileDataTypes` here?

File: src/types/profileData.types.js (hunk 3 above)
Comment (andriiwp3) [reply src/types/profileData.types.js:11]: @AndriyTykhan You mean `profileType`?

Comment (AndriyTykhan) [reply src/types/profileData.types.js:11]: File name and prop type

Comment (andriiwp3) [reply src/types/profileData.types.js]: @AndriyTykhan done ✅

Comment (AndriyTykhan) [reply src/types/profileData.types.js]: Or we can name type as `profileType`

Comment (andriiwp3) [reply src/types/profileData.types.js]: Will push (force) for a 2 minutes

**Review by AndriyTykhan — APPROVED** [review]

File: src/types/profileData.types.js (hunk 3 above)
Comment (andriiwp3) [reply src/types/profileData.types.js]: pushed

**Review by AndriyTykhan — APPROVED** [review]

**Review by bohdan-ukr — APPROVED** [review]

**Pull request merged**